app/controllers/application_controller.rb

```diff
@@ -6,8 +6,21 @@ class ApplicationController < ActionController::API
   # protect_from_forgery with: :exception
 
   def authenticated?
-    authenticate_or_request_with_http_basic do
-      |u, p| User.where(username: u, password: p).present?
+    authenticate_or_request_with_http_basic do |u, p|
+      @current_user = User.find_by(username: u, password: p)
     end
   end
+
+  def current_user
+    @current_user
+  end
+
+  def error(status, message = 'Something went wrong')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.